client/matching/client.go

```diff
@@ -258,24 +258,33 @@ func (c *clientImpl) GetTaskListsByDomain(
 		futures = append(futures, future)
 	}
 
-	taskListMap := make(map[string]struct{})
+	decisionTaskListMap := make(map[string]*types.DescribeTaskListResponse)
+	activityTaskListMap := make(map[string]*types.DescribeTaskListResponse)
 	for _, future := range futures {
 		var resp *types.GetTaskListsByDomainResponse
 		if err = future.Get(ctx, &resp); err != nil {
 			return nil, err
 		}
-		for _, tl := range resp.TaskListNames {
-			if _, ok := taskListMap[tl]; !ok {
-				taskListMap[tl] = struct{}{}
+		for name, tl := range resp.GetDecisionTaskListMap() {
+			if _, ok := decisionTaskListMap[name]; !ok {
+				decisionTaskListMap[name] = tl
+			} else {
+				decisionTaskListMap[name].Pollers = append(decisionTaskListMap[name].Pollers, tl.GetPollers()...)
```

Comment:
It would be great if we can add some unit tests for client implementation that has special logic.

Reply:
We don't have any right now. I created a task to cover this folder.